src/frontends/src/query_frontend.hpp

```diff
@@ -479,22 +489,33 @@ class RequestHandler {
    * JSON format for prediction query request:
    * {
    *  "input" := [double] | [int] | [string] | [byte] | [float]
+   *  "input_batch" := [[double] | [int] | [byte] | [float] | string]
    * }
    */
-  folly::Future<Response> decode_and_handle_predict(
+  folly::Future<std::vector<folly::Try<Response>>> decode_and_handle_predict(
       std::string json_content, std::string name,
       std::vector<VersionedModelId> models, std::string policy,
       long latency_slo_micros, InputType input_type) {
     rapidjson::Document d;
     clipper::json::parse_json(json_content, d);
     long uid = 0;
-    // NOTE: We will eventually support personalization again so this commented
-    // out code is intentionally left in as a placeholder.
-    // long uid = clipper::json::get_long(d, "uid");
-    std::shared_ptr<Input> input = clipper::json::parse_input(input_type, d);
-    auto prediction = query_processor_.predict(
-        Query{name, uid, input, latency_slo_micros, policy, models});
-    return prediction;
+    std::vector<folly::Future<Response>> predictions;
+
+    if (d.HasMember("input")) {
+      std::shared_ptr<Input> input = clipper::json::parse_input(input_type, d);
+      auto prediction = query_processor_.predict(
+          Query{name, uid, input, latency_slo_micros, policy, models});
+      predictions.push_back(std::move(prediction));
+    } else {  // d.HasMember("input_batch") instead
```

Comment:
Does it make sense to explicitly check for `d.hasMember("input_batch")` here, rather than leaving it as an else statement?